Realm/RLMConstants.h

```diff
@@ -18,6 +18,14 @@
 
 #import <Foundation/Foundation.h>
 
+// For compatibility with Xcode 7, before extensible string enums were introduced,
+#ifdef NS_EXTENSIBLE_STRING_ENUM
+#define RLM_EXTENSIBLE_STRING_ENUM_CASE_SWIFT_NAME(_, extensible_string_enum) NS_SWIFT_NAME(extensible_string_enum)
+#else
+#define NS_EXTENSIBLE_STRING_ENUM
```

Comment:
I don't think we want to define this macro in our public headers. It'll break other code doing a similar #ifdef check on it.